homeassistant/components/volvooncall/binary_sensor.py

```diff
@@ -18,22 +18,35 @@ async def async_setup_platform(
     """Set up the Volvo sensors."""
     if discovery_info is None:
         return
-    async_add_entities([VolvoSensor(hass.data[DATA_KEY], *discovery_info)])
+    async_add_entities([VolvoSensor(hass, *discovery_info)])
 
 
 class VolvoSensor(VolvoEntity, BinarySensorEntity):
     """Representation of a Volvo sensor."""
 
-    @property
-    def is_on(self):
-        """Return True if the binary sensor is on, but invert for the 'Door lock'."""
-        if self.instrument.attr == "is_locked":
-            return not self.instrument.is_on
-        return self.instrument.is_on
+    def __init__(
+        self, hass: HomeAssistant, vin, component, attribute, slug_attr, coordinator
+    ):
+        """Initialize the sensor."""
+        VolvoEntity.__init__(
+            self, hass, vin, component, attribute, slug_attr, coordinator
+        )
 
-    @property
-    def device_class(self):
-        """Return the class of this sensor, from DEVICE_CLASSES."""
         if self.instrument.device_class in DEVICE_CLASSES:
-            return self.instrument.device_class
-        return None
+            self._attr_device_class = self.instrument.device_class
+        else:
+            self._attr_device_class = None
+
+        if self.instrument.attr == "is_locked":
+            self._attr_is_on = not self.instrument.is_on
+        else:
+            self._attr_is_on = self.instrument.is_on
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
```

Comment:
Implement the `is_on` property instead. It will save lines and consolidate this code.